lib/northbound.c

```diff
@@ -1867,12 +2069,15 @@ const char *nb_event_name(enum nb_event event)
 	case NB_EV_APPLY:
 		return "apply";
 	default:
-		return "unknown";
+		snprintf(buf, sizeof(buf), "unknown(%u)", event);
+		return buf;
 	}
 }
 
 const char *nb_operation_name(enum nb_operation operation)
 {
+	static char buf[32];
```

Comment:
Are there any thread safety issues to think about here? Same for the rest of them

Reply:
I'd say no since the NB API can only be used safely from the main pthread (and I can't see this changing anytime soon, if ever).

I wouldn't worry about making these functions reentrant either since their usage scope is very limited. My understanding is that requiring extra `buf/buf_len` parameters would hurt code readability for no real benefit.

We could also add new printfrr format specifiers for these NB types, but that would be an overkill IMO.